### PR TITLE
Base for fields definition

### PR DESCRIPTION
**Pull request by postatum**

### CODE REVIEW

File: tests/test_fields.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import datetime
+
+import pytest
+from elasticsearch_dsl.exceptions import ValidationException
+
+from nefertari_es import fields
+
+
+class TestFielfHelpers(object):
```
Comment (latteier) [inline tests/test_fields.py]: typo.

Comment (postatum) [inline tests/test_fields.py]: fixed

File: nefertari_es/fields.py [hunk 2]
```diff
@@ -1,30 +1,148 @@
...
+    def __init__(self, *args, **kwargs):
+        self._primary_key = kwargs.pop('primary_key', False)
+        super(BaseFieldMixin, self).__init__(*args, **kwargs)
+
+
+class IdField(CustomMappingMixin, field.String):
+    name = 'idfield'
+    _custom_mapping = {'type': 'string'}
+
+    # TODO: Mirror document.meta['_id'] value
```
Comment (latteier) [inline nefertari_es/fields.py]: How will we do this? I'm not sure if the field has access to the model. I need to investigate this more.

Comment (postatum) [inline nefertari_es/fields.py]: No idea yet. I thought about using descriptor, but elasticsearch-dsl has weird logic for attributes access so descriptors wont work probably.

**Pull request merged**